test/e2e/route_metrics_test.go

```diff
@@ -0,0 +1,665 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	routev1client "github.com/openshift/client-go/route/clientset/versioned"
+	"github.com/openshift/library-go/test/library/metrics"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+// TestRouteMetricsControllerOnlyRouteSelector creates an Ingress Controller with only RouteSelector and creates
+// a Route with a label matching the RouteSelector. The Ingress Controller is modified to change the RouteSelector
+// and then again the RouteSelector is changed back to the original value. The Route is then deleted and after that
+// the Ingress Controller is deleted. For all the events (Create, Update, Delete), the metrics is verified with the
+// expected metrics value.
+func TestRouteMetricsControllerOnlyRouteSelector(t *testing.T) {
```

Comment:
Thanks so much for creating the e2e tests for this. Looks good except a few nits.

Can you change this so there is one function that works on 3 test cases, instead of one function per test case?  Then, if we should need to add a test case in the future, we only need to add support for a new test case.

Reply:
Created a common function and called it from the 3 different functions with appropriate boolean values as parameters for the common function. Any changes can be made in the common function for further addition of test cases.